joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/TypeInferencePass.scala

```diff
@@ -12,73 +12,76 @@ import io.shiftleft.semanticcpg.language._
-    namePartsCache.get(fullName).toScala.getOrElse {
-      val nameParts = fullName match {
-        case NamePartsPattern(maybeTd, maybeSig) =>
-          val typeDecl  = Option(maybeTd).map(_.init)
-          val signature = Option(maybeSig)
-          NameParts(typeDecl, signature)
+  private def getNameParts(name: String, fullName: String): NameParts = {
+    val Array(qualifiedName, signature) = fullName.split(":", 2)
 
-        case _ =>
-          logger.warn(s"Could not extract name parts from $fullName")
-          NameParts(None, None)
-      }
-      namePartsCache.put(fullName, nameParts)
-      nameParts
+    val typeDeclName = qualifiedName.stripSuffix(name) match {
+      case "" => None
+
+      case typeDeclName => Some(typeDeclName)
     }
+
+    println(s"Got name parts ${typeDeclName}, ${signature}")
+    NameParts(typeDeclName, signature)
   }
 
   private def getReplacementMethod(call: Call): Option[Method] = {
     cache.get(call.methodFullName).toScala.getOrElse {
-      val callNameParts    = getNameParts(call.methodFullName)
-      val candidateMethods = cpg.method.internal.nameExact(call.name)
-
-      val uniqueMatchingMethod = candidateMethods.find(isMatchingMethod(_, call, callNameParts)).flatMap { method =>
-        val otherMatchingMethod = candidateMethods.find(isMatchingMethod(_, call, callNameParts))
-        // Only return a resulting method if exactly one matching method is found.
-        Option.when(otherMatchingMethod.isEmpty)(method)
+      val callNameParts = getNameParts(call.name, call.methodFullName)
+      resolvedMethodIndex.get(call.name).flatMap { candidateMethods =>
+        val candidateMethodsIter = candidateMethods.iterator
+        val uniqueMatchingMethod =
+          candidateMethodsIter.find(isMatchingMethod(_, call, callNameParts)).flatMap { method =>
+            val otherMatchingMethod = candidateMethodsIter.find(isMatchingMethod(_, call, callNameParts))
+            // Only return a resulting method if exactly one matching method is found.
+            Option.when(otherMatchingMethod.isEmpty)(method)
+          }
+        cache.put(call.methodFullName, uniqueMatchingMethod)
+        uniqueMatchingMethod
       }
-      cache.put(call.methodFullName, uniqueMatchingMethod)
-      uniqueMatchingMethod
     }
   }
-  override def runOnPart(diffGraph: DiffGraphBuilder, call: Call): Unit = {
-    getReplacementMethod(call).foreach { method =>
-      diffGraph.setNodeProperty(call, PropertyNames.MethodFullName, method.fullName)
-      diffGraph.setNodeProperty(call, PropertyNames.Signature, method.signature)
-      diffGraph.setNodeProperty(call, PropertyNames.TypeFullName, method.methodReturn.typeFullName)
+
+  override def runOnPart(diffGraph: DiffGraphBuilder, part: (String, List[Call])): Unit = {
+    part match {
+      case (_, Nil)      => // This should never happen
+      case (name, calls) =>
+        // Since call full names match, the replacement method for each will be the same
```

Comment:
The full names match but only because they contain "unknown signature" so why is it the case that we are allowed to treat the call sites the same?

Reply:
Ah, I did miss the fact that this skips argument type checking. I'll update the PR to handle this better.